src/mds/MDCache.cc

```diff
@@ -880,8 +880,9 @@ void MDCache::adjust_subtree_auth(CDir *dir, mds_authority_t auth)
 void MDCache::try_subtree_merge(CDir *dir)
 {
   dout(7) << "try_subtree_merge " << *dir << dendl;
-  assert(subtrees.count(dir));
-  set<CDir*> oldbounds = subtrees[dir];
+  auto it = subtrees.find(dir);
+  assert(it != subtrees.end());
```

Comment:
Better would be `auto &oldbounds = subtrees.at(dir)`? This would obviate the assert.